sentry-android-core/src/main/java/io/sentry/android/core/SendCachedEnvelopeIntegration.java

```diff
@@ -75,7 +75,9 @@ public void close() throws IOException {
   @Override
   public void onConnectionStatusChanged(
       final @NotNull IConnectionStatusProvider.ConnectionStatus status) {
-    if (scopes != null && options != null) {
+    if (scopes != null
+        && options != null
+        && status != IConnectionStatusProvider.ConnectionStatus.DISCONNECTED) {
```

Comment:
kind of unrelated to this PR, but I noticed that we also submit a task even if the status was DISCONNECTED, which was unnecessary since we'd anyway bail out in the submitted task itself.